src/controllers/servicesController.ts

```diff
@@ -34,6 +39,65 @@ export default class ServicesController {
     | IResponseSuccessJson<ServicePublic>
   > => this.messagesService.getService(req.params.id);
 
+  /**
+   * Returns the service preferences for the provided service id
+   */
+  public readonly getServicePreferences = (
+    req: express.Request
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseErrorConflict
+    | IResponseErrorTooManyRequests
+    | IResponseSuccessJson<ServicePreference>
+  > =>
+    withUserFromRequest(req, async user =>
+      withValidatedOrValidationError(
+        ServiceId.decode(req.params.id),
+        serviceId =>
+          this.messagesService.getServicePreferences(
+            user.fiscal_code,
+            serviceId
+          )
+      )
+    );
+
+  /**
+   * Create or Update the service preferences for the provided service id
+   */
+  public readonly upsertServicePreferences = (
+    req: express.Request
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseErrorConflict
+    | IResponseErrorTooManyRequests
+    | IResponseSuccessJson<ServicePreference>
+  > =>
+    withUserFromRequest(req, async user =>
+      withValidatedOrValidationError(
+        ServiceId.decode(req.params.id),
+        serviceId =>
+          withValidatedOrValidationError(
+            ServicePreference.decode(req.body.body),
```

Comment:
Why a double body? This is the body from the express request.
```suggestion
            ServicePreference.decode(req.body),
```

Reply:
Right, sorry..